### PR TITLE
refactor: drop async pipe in `tui-dialogs`

### PR DESCRIPTION
**Pull request by splincode**

### CODE REVIEW

File: projects/core/components/dialog/dialogs.component.ts [hunk 1]
```diff
@@ -33,6 +33,9 @@ import {TUI_DIALOGS} from './dialog.tokens';
 })
 export class TuiDialogs {
     protected readonly el = tuiInjectElement();
-    protected readonly dialogs$: Observable<ReadonlyArray<TuiPopover<any, any>>> =
-        inject(TUI_DIALOGS);
+    protected readonly dialogs = toSignal<ReadonlyArray<TuiPopover<any, any>>>(
```
Comment (vladimirpotekhin) [inline projects/core/components/dialog/dialogs.component.ts:36]: it looks like the overlay is now blinking

https://github.com/user-attachments/assets/fb2a8eba-4769-4c7e-bfdf-34740f51fec1

Comment (splincode) [reply projects/core/components/dialog/dialogs.component.ts:37]: fixed

**Review by vladimirpotekhin — APPROVED** [review]

File: projects/core/components/dialog/dialogs.template.html [hunk 2]
```diff
@@ -1,4 +1,4 @@
-<ng-container *ngIf="dialogs$ | async as dialogs">
+<ng-container *ngIf="dialogs() as dialogs">
```
Comment (waterplea) [inline projects/core/components/dialog/dialogs.template.html:1]: Let's drop the container and just use `dialogs()` multiple times.

Comment (splincode) [reply projects/core/components/dialog/dialogs.template.html:1]: @waterplea https://github.com/taiga-family/taiga-ui/pull/8913#discussion_r1750107775

**Review by waterplea — APPROVED** [review]

**Pull request closed without merging**